### PR TITLE
Explain what usage "search" is for Intl.Collator and caution against misuse

### PR DESCRIPTION
**Pull request by hsivonen**
Also fixes a couple of adjacent remarks.

### Description

Explains what `usage: "search"` for `Intl` collator is for and cautions against misuse.

### Motivation

The purpose of this API feature isn't well known. It would have helped me previously to have this explanation instead of having to seek it from an ECMA-402 committee meeting.

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/intl/collator/collator/index.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Intl.Collator(locales, options)
         - `phonetic` (Lingala)
         - `pinyin` (Chinese)
         - `reformed` (formerly Swedish; do not specify explicitly as this was the old name for the default for Swedish)
-        - `searchjl` (Korean; do not use as this is for searching rather than sorting, and the API covers only sorting)
+        - `searchjl` (Korean; not for sorting; for search matching such that syllable-initial consonants are matched and vowels and possible syllable-final consonants are ignored)
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/intl/collator/collator/index.md:56]:
```suggestion
        - `searchjl` (Korean; this is not for sorting, but for search matching such that syllable-initial consonants are matched and vowels and possible syllable-final consonants are ignored)
```

File: files/en-us/web/javascript/reference/global_objects/intl/collator/collator/index.md [hunk 2]
```diff
@@ -137,12 +134,12 @@ Intl.Collator(locales, options)
...
         - `emoji` (root)
         - `eor` (root)
-        - `gb2312` (Chinese; do not use; not available in Chrome or Edge)
+        - `gb2312` (Chinese; do not use; not available in Firefox, Chrome or Edge)
         - `phonebk` (German)
         - `phonetic` (Lingala)
         - `pinyin` (Chinese)
         - `reformed` (formerly Swedish; do not specify explicitly as this was the old name for the default for Swedish)
-        - `searchjl` (Korean; do not use as this is for searching rather than sorting, and the API covers only sorting)
+        - `searchjl` (Korean; not for sorting; for search matching such that syllable-initial consonants are matched and vowels and possible syllable-final consonants are ignored)
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/reference/global_objects/intl/collator/collator/index.md:142]:
```suggestion
        - `searchjl` (Korean; this is not for sorting, but for search matching such that syllable-initial consonants are matched and vowels and possible syllable-final consonants are ignored)
```

**Review by Josh-Cena — COMMENTED** [review]
I have a question, since I'm not super familiar with Intl: `sort` seems like a superset of `search`. When used with the same `sensitivity`, when would one prefer `search` over `sort`?

**Comment by hsivonen** [comment]
> I have a question, since I'm not super familiar with Intl: `sort` seems like a superset of `search`. When used with the same `sensitivity`, when would one prefer `search` over `sort`?

One would use `search` when filtering a list of items down to only items matched by a search key. It's a very narrow use case due to the API only allowing full-string matching and not substring matching. (So narrow that I think it doesn't make sense to expose an API for it, but it's already exposed.)

Compared to just using `sort` for this case:
* A couple of languages differ by leaving accents _not_ ignored in _more_ cases in the accent-_insensitive_ mode.
* A couple of languages use a definition that on the `sort` side isn't the default one of multiple options available for sorting.
* A couple of scripts become insensitive to some things that are visually accent-like even if not technically accents.
* For many languages, the equal vs. not-equal result doesn't change even if the order would change if `search` was misused for sorting.
* Plus a couple of other differences that I don't want to document, because I want to leave the option open to change them.

The main point I want to make is that it's incorrect to use `usage: "search"` for sorting, i.e. in a way that gives user-visible distinction to non-zero return values of `compare()`. (The API would be more misuse-resistant if `usage: "search"` used a different object that had a method called `matches()` returning a boolean.)

**Comment by Josh-Cena** [comment]
I understand you cannot use `search` for sorting, but I wonder if you can use `sort` for searching. From what I understand, you are saying that sort tends to be more granular than `search` even when all other options are the same, so the search may not find what one wants?

**Comment by hsivonen** [comment]
> I understand you cannot use `search` for sorting, but I wonder if you can use `sort` for searching. From what I understand, you are saying that `sort` tends to be more granular than `search` (i.e. less likely to compare equal) even when all other options are the same, so the search may not find what one wants?

`sort` is generally more granular than `search`, but for Catalan, things go the other way. In general, it would be bad to suggest to readers that they can expect such specific things.

Meanwhile, there's been a spec PR merged on this topic: https://github.com/tc39/ecma402/pull/792

I'll get the merge conflicts addressed.

**Comment by Josh-Cena** [comment]
Okay, I think this is definitely an improvement from the existing content.

@ben-allen—if this change looks good to you, we can merge it.

**Comment by ben-allen** [comment]
LGTM!

**Review by Josh-Cena — APPROVED** [review]
Great! I have no intention to block it further because I don't have good suggestions. Thanks a lot @hsivonen and @ben-allen—your professional knowledge is exactly what's needed :)

**Pull request merged**